Common/Model.cs

```diff
@@ -0,0 +1,143 @@
+            LogStream logstream = new LogStream(delegate (String msg, String userData)
+            {
+                Console.WriteLine(msg);
+            });
+            logstream.Attach();
+
+            //Import the model. All configs are set. The model
+            //is imported, loaded into managed memory. Then the unmanaged memory is released, and everything is reset.
+            Scene scene = importer.ImportFile(path, PostProcessSteps.Triangulate);
+
+            // check for errors
+            if (scene == null || scene.SceneFlags.HasFlag(SceneFlags.Incomplete) || scene.RootNode == null)
+            {
+                Console.WriteLine("Unable to load model from: " + path);
+                return;
+            }
+            
+            //Reset the meshes and textures
+            meshes = new List<Mesh>();
+
+            //Set the scale of the model
+            float scale = 1/200.0f;
+            Matrix4x4 scalingMatrix = new Matrix4x4(scale, 0, 0, 0, 0, scale, 0, 0, 0, 0, scale, 0, 0, 0, 0, 1);
+
+            // process ASSIMP's root node recursively. We pass in the scaling matrix as the first transform
+            ProcessNode(scene.RootNode, scene, scalingMatrix);
+
+            importer.Dispose();
+        }
+
+        // processes a node in a recursive fashion. Processes each individual mesh located at the node and repeats this process on its children nodes (if any).
+        private void ProcessNode(Node node, Scene scene, Matrix4x4 parentTransform)
+        {
+            //Multiply the transform of each node by the node of the parent, this will place the meshes in the correct relative location
+            Matrix4x4 transform = node.Transform*parentTransform;
+
+            // process each mesh located at the current node
+            for (int i = 0; i < node.MeshCount; i++)
+            {
+                // the node object only contains indices to index the actual objects in the scene. 
+                // the scene contains all the data, node is just to keep stuff organized (like relations between nodes).
+                AssimpMesh mesh = scene.Meshes[node.MeshIndices[i]];
+                meshes.Add(ProcessMesh(mesh, transform));
+            }
+
+            for (int i = 0; i < node.ChildCount; i++)
+            {
+                ProcessNode(node.Children[i], scene, transform);
+            }
+        }
+
+        private Mesh ProcessMesh(AssimpMesh mesh, Matrix4x4 transform)
+        {
+            // data to fill
+            List<Vertex> vertices = new List<Vertex>();
+            List<int> indices = new List<int>();
+
+            // walk through each of the mesh's vertices
+            for (int i = 0; i < mesh.VertexCount; i++)
+            {
+                Vertex vertex = new Vertex();
+                // we declare a placeholder vector since assimp uses its own vector class that doesn't directly convert to the Vector3 class so we transfer the data to this placeholder Vector3 first.
```

Comment:
We can probably do a reinterpret cast here or something similar so that is no problem.

Reply:
See the comments above, but also maybe this whole vertex creation process could all just be put into a reinterpret cast. I can look into that more if we don't like what I have done converting to OpenTk types. Note that a lot of this code is coming pretty much straight out of [learnopengl.com](https://learnopengl.com/Model-Loading/Model), so it would make sense that it looks a little too much like C++ and not enough like C#.